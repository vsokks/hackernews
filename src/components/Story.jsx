import {Col, Container, Row} from "react-bootstrap";
import {convertDateToString} from "../utils/DateUtils";

const Story = ({story}) => {
    return (
        <Container fluid="xs" className="card" style={{padding: "5px", marginTop: "8px", marginBottom: "8px", borderColor: "darkgrey"}}>
            <Row xs={12}>
                <Col>
                    <h6 className="card-title">
                        {story.url && <a href={story.url} target="_blank" rel="noreferrer">{story.title}</a>}
                        {!story.url && story.title}
                    </h6>
                </Col>
            </Row>
            <Row>
                <Col xs={12}>
                    <p className="card-text">
                        By <i>{story.by}</i> on {convertDateToString(story.time)}
                    </p>

                </Col>
            </Row>
        </Container>
    );
}
export default Story;