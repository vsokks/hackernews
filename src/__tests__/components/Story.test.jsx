import React from 'react';
import renderer from 'react-test-renderer';
import Story from "../../components/Story";

test('Test Story Component with hyper link URL', () => {
    let story = {
        title : 'This is sample title',
        by: 'test-user',
        time: 1207886576
    } ;
    const component = renderer.create(<Story story={story} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});

test('Test Story Component without hyper link URL', () => {
    let story = {
        url : 'www.test.com',
        title : 'This is sample title',
        by: 'test-user'
    } ;
    const component = renderer.create(<Story story={story} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});