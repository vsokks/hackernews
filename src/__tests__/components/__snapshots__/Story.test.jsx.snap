// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Story Component with hyper link URL 1`] = `
<div
  className="card container-xs"
  style={
    Object {
      "borderColor": "darkgrey",
      "marginBottom": "8px",
      "marginTop": "8px",
      "padding": "5px",
    }
  }
>
  <div
    className="row row-cols-12"
  >
    <div
      className="col"
    >
      <h6
        className="card-title"
      >
        This is sample title
      </h6>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <p
        className="card-text"
      >
        By 
        <i>
          test-user
        </i>
         on 
        4/11/2008, 9:32 AM
      </p>
    </div>
  </div>
</div>
`;

exports[`Test Story Component without hyper link URL 1`] = `
<div
  className="card container-xs"
  style={
    Object {
      "borderColor": "darkgrey",
      "marginBottom": "8px",
      "marginTop": "8px",
      "padding": "5px",
    }
  }
>
  <div
    className="row row-cols-12"
  >
    <div
      className="col"
    >
      <h6
        className="card-title"
      >
        <a
          href="www.test.com"
          rel="noreferrer"
          target="_blank"
        >
          This is sample title
        </a>
      </h6>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <p
        className="card-text"
      >
        By 
        <i>
          test-user
        </i>
         on 
      </p>
    </div>
  </div>
</div>
`;
